CheckBreach_And_AlertTarget.c

```diff
@@ -0,0 +1,82 @@
+#include "typewise-alert.h"
+#include <stdio.h>
+
+Successtype (*AlertTargetFunc[])(BreachType breachType)={	sendToController,sendToEmail, sendToConsole};
+
+unsigned int SentSuccessfullytoController=0;
+/**
+ ***************************************************************************************************
+ * Function Name: checkAndAlert 
+ * 
+ * Function Description: Checks the temperature breach based on cooling type and temeperature in degree celcius and alerts to all the target devices.
+ *
+ * \param  Inputs:- AlertTarget alertTarget
+ *					BatteryCharacter batteryChar
+ *					double temperatureInC
+ *					
+ *		   Outputs:- None
+ *         
+ * \return  Successtype  (ENUM) 
+ *          
+ * \retval  CheckAndAlertSuccess:- Succcess parameter which points to Successtype ENUM based on alert target.
+ ***************************************************************************************************
+ */
+
+Successtype checkAndAlert(AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) 
+{
+   BreachType breachType = classifyTemperatureBreach(batteryChar.coolingType, temperatureInC);
+   Successtype CheckAndAlertSuccess =(*AlertTargetFunc[alertTarget])(breachType);
+   return CheckAndAlertSuccess;
+}
+
+void DioWriteToController(const unsigned short header, BreachType breachType)
+{
+	(void)header;
+	(void)breachType;
+    SentSuccessfullytoController = 1;

```

Comment:
`SentSuccessfullytoController` is set here but not checked anywhere? So even if this is not done, or email is called in place of controller by mistake, the tests will still pass.
So even after the tests pass, your functionality will still need to be tested in an integrated environment.
Any way to strengthen the test? Why not write a REQUIRE on `SentSuccessfullytoController`?